src/modules/tools/endstops/Endstops.cpp

```diff
@@ -669,15 +674,14 @@ void Endstops::on_gcode_received(void *argument)
 
                 bool has_endstop_trim = this->is_delta || this->is_scara;
                 if (has_endstop_trim) {
-                    float ideal_actuator_position[3];
+                    ActuatorCoordinates ideal_actuator_position;
                     THEKERNEL->robot->arm_solution->cartesian_to_actuator(ideal_position, ideal_actuator_position);
 
                     // We are actually not at the ideal position, but a trim away
-                    float real_actuator_position[3] = {
-                        ideal_actuator_position[X_AXIS] - this->trim_mm[X_AXIS],
-                        ideal_actuator_position[Y_AXIS] - this->trim_mm[Y_AXIS],
-                        ideal_actuator_position[Z_AXIS] - this->trim_mm[Z_AXIS]
-                    };
+                    ActuatorCoordinates real_actuator_position;
+                    for (size_t i = THEKERNEL->robot->actuators.size() + 1; i--;) {
```

Comment:
this should be remain hard coded to 3, no changes until it is understood how to handle more than 3 motors for endstops. FWIW this is mostly used for deltas which would haeve no reason to have more than 3 motors.